tests/baselines/reference/mappedTypeConstraints2.errors.txt

```diff
@@ -64,8 +64,16 @@ mappedTypeConstraints2.ts(82,9): error TS2322: Type 'ObjectWithUnderscoredKeys<K
       let s: `_${string}` = obj[key]; // Error
           ~
 !!! error TS2322: Type 'Mapped6<K>[keyof Mapped6<K>]' is not assignable to type '`_${string}`'.
-!!! error TS2322:   Type 'Mapped6<K>[string] | Mapped6<K>[number] | Mapped6<K>[symbol]' is not assignable to type '`_${string}`'.
-!!! error TS2322:     Type 'Mapped6<K>[string]' is not assignable to type '`_${string}`'.
+!!! error TS2322:   Type 'Mapped6<K>[`_${K}`]' is not assignable to type '`_${string}`'.
+!!! error TS2322:     Type 'Mapped6<K>[`_${string}`]' is not assignable to type '`_${string}`'.

```

Comment:
I believe this change is a fix. The new error matches what was reported by TS 5.0: [TS playground](https://www.typescriptlang.org/play/?ts=5.0.4#code/C4TwDgpgBAsghmSATAbAHgNJQgD2BAOyQGcpjgAnASwIHMA+KAXigG8AoKKAbQAUoaULHFIADAPoASVrwC+ogLoAuKLwDc7WRvYAzAK4EAxsCoB7AlB3osufEVLlqdegApTAIwBWK+IgipMegAaKABrCBAVcJBTHVgEZGt6AEo2TigAGwhgMhUJaUcaWnlmKA9PbmiFNSgAelqoAFEKClMKTXYgA)